### PR TITLE
docs(en): update v5 migrate guide and container api reference

### PR DESCRIPTION
**Pull request by xsynaptic**
#### Description (required)

- Update MDX update section of the v5 guide to include a notice about MDX server renderers
- Update Astro Container API docs to include new reference
- Also update the v5 guide edit link to point to itself

#### Related issues & labels (optional)

- Closes #
- Suggested label:

### CODE REVIEW

**Review by Adammatthiesen — APPROVED** [review]
Looks right to me - Just a random guy who spent the last week staring at all those files in core....

File: src/content/docs/en/reference/container-reference.mdx [hunk 1]
```diff
@@ -97,7 +97,7 @@ The following example manually imports the necessary server renderers to be able
 ```js
 import reactRenderer from "@astrojs/react/server.js";
 import vueRenderer from "@astrojs/vue/server.js";
-import mdxRenderer from "astro/jsx/server.js";
+import mdxRenderer from "@astrojs/mdx/server.js";
```
Comment (sarah11918) [inline src/content/docs/en/reference/container-reference.mdx:100]: Nice spotting, thank you!

File: src/content/docs/en/guides/upgrade-to/v5.mdx [hunk 2]
```diff
@@ -168,6 +168,13 @@ Astro v5.0 moves this responsibility to handle and render JSX and MDX to the `@a
 
 If your project includes `.mdx` files, you must upgrade `@astrojs/mdx` to the latest version (v4.0.0) so that your JSX can be handled properly by the integration.
 
+If you are using an MDX server renderer with the experimental [Astro Container API](/en/reference/container-reference/) you should update the import:
```
Comment (sarah11918) [inline src/content/docs/en/guides/upgrade-to/v5.mdx:171]:
```suggestion
If you are using an MDX server renderer with the experimental [Astro Container API](/en/reference/container-reference/) you must update the import to reflect the new location:
```

**Review by sarah11918 — APPROVED** [review]
This looks great! Sorry it was a busy end to the week last week and it took me a few days to get to this, but this does the job nicely! Welcome to Team Docs! :partying_face:

**Pull request merged**